src/main.rs

```diff
@@ -35,6 +35,10 @@ fn main(mut req: Request) -> Result<Response, Error> {
         // Deny anything else.
         _ => {
             return Ok(Response::from_status(StatusCode::METHOD_NOT_ALLOWED)
+                .with_header(
+                    header::ALLOW,
+                    format!("{}, {}, {}", Method::GET, Method::HEAD, "PURGE"),
```

Comment:
In this particular case wouldn't be better to simply use the literal "GET, HEAD, PURGE" ?

Reply:
My inclination is to say that the compiler ought to be smart enough to figure this out and emit some equivalent code when compiled with `--release`. Because I just _had_ to know, I did some digging!

```rust
/// Entrypoint.
///
/// Compile this program with `cargo build --release --target=wasm32-wasi`.
///
/// Use `wabt`'s `wasm2wat` utility to convert the binary into WebAssembly
/// text format for further inspection.
fn main() {
    // Create two strings, providing either `T: Display` arguments, or a
    // string literal. Make sure they are differentiable. (123456 vs. 654321)
    let format_with_args = format!("{}, {}", 123456u32, "PURGE");
    let no_args = format!("654321, PURGE");

    // Print the strings out, so we know they won't be optimized away.
    println!("{} {}", format_with_args, no_args);
}
```

If you grep through the `.wat` generated by this program, you'll see in the `(data ...)` section that there are two strings with `PURGE`, so I feel confident saying that this code should be okay as is :slightly_smiling_face:

Reply:
Isn't this allocating whether the non format! variant is not though?

Reply:
In the wasm file I can see a lot of extra calls for the format! version, since it needs to get the Display-ed representation, allocating the string, formatting it, deallocating the string, etc.

Is there any reason you will prefer that over using literals?

Reply:
Using `format!` seems slightly more clumsy, and possibly more typo-prone.

I also do not believe it is any real violation of the spec to not publicize `PURGE` in a response like this.  Plenty of Fastly docs imply that you may send a PURGE to almost any resource on a Fastly-hosted page, and in practice that isn't a huge problem, but I think it is a step too far to actually advertise that fact on the wire like this.

I'd just do `"GET, HEAD"` :)

Reply:
Ok am gonna go with @jeddenlea's suggestion as agree that PURGE shouldn't be advertised like this.

Reply:
Fixed in https://github.com/fastly/compute-starter-kit-rust-default/pull/30/commits/f3a62d5fae915b6a592f78356b34841eb99cfc69